client_runner.rb

```
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+require 'httparty'
+require 'digest'
+
+module QuikNode
+  module API
+    class Base
+      include HTTParty
+      base_uri 'https://quiknode.pro'
+
+      def details(id:, referrer_only: false, admin_token: nil)
+        {
+          'http_provider' => "https://qn-fullstack-proxy.herokuapp.com/#{admin_token}/"
+        }
+      end
+
+      def last_block(ex_id:, token:)
+        params = {
+          jsonrpc: '2.0',
+          method: 'eth_blockNumber',
+          params: [],
+          id: 1,
+        }
+        resp = self.class.post(
```

There are couple errors that could happen here, and it would be nice to treat them separately to guarantee we have the right level of visibility in case something happens in production.

The two main errors:

- The request has failed and an exception is thrown (authorization)
- The request is successful but the response cannot be parsed (e.g. "result") does not exist.

In both cases, do we want to return an error?